src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,279 @@
+      host: applicativeHostUrl,
+      method: requestMethod,
+      authenticationField: authentication,
+    }
+
+    this.bodyMaxLength = bodyMaxLength
+    this.acceptUnconvertedRows = acceptUnconvertedRows
+
+    this.mapping = mapping || {}
+    this.csvDelimiter = csvDelimiter
+    this.proxy = this.getProxy(proxy)
+
+    this.canHandleFiles = true
+    this.canHandleValues = false
+  }
+
+  /**
+   * Send the file.
+   * @param {String} filePath - The path of the file
+   * @return {Promise} - The send status
+   */
+  async handleFile(filePath) {
+    // Verify that the file receive is a csv one
+    const regexExp = RegExp('.csv$')
+
+    if (!regexExp.test(filePath)) {
+      this.logger.error(`Invalid file format (.csv file expected), file (${filePath} skipped`)
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    const csvDataParsed = await this.parseCsvFile(filePath)
+
+    if (csvDataParsed.length === 0) {
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    // The csv parsing is a success so we begin to map the content
+    // Initialize the body to an empty array
+    const httpBody = this.convertToHttpBody(csvDataParsed)
+
+    if (httpBody.length !== 0) {
+      if (!await this.sendData(httpBody)) {
+        this.logger.error('Impossible to send data')
+        return ApiHandler.STATUS.LOGIC_ERROR
+      }
+    }
+
+    // Logs all the erros
+    if (convertionErrorBuffer.error.length > 0) {
+      this.logger.error(`${convertionErrorBuffer.error.length} convertions error`)
+
+      if (convertionErrorBuffer.error.length > ERROR_PRINT_SIZE) {
+        for (let i = 0; i < ERROR_PRINT_SIZE; i += 1) {
+          this.logger.error(`${convertionErrorBuffer.error[i]}`)
+        }
+      } else {
+        convertionErrorBuffer.error.forEach((index) => {
+          this.logger.error(`${convertionErrorBuffer.error[index]}`)
+        })
+      }
+    }
+
+    return ApiHandler.STATUS.SUCCESS
+  }
+
+  /**
+   * @param {string} filePath - Path of the file to parse
+   * @return {Promise} - the parsed content of the file
+   */
+  parseCsvFile(filePath) {
+    return new Promise((resolve, reject) => {
+      // Initialize array wich will be filled at each step
+      const csvDataParsed = []
+      const csvFile = fs.createReadStream(filePath)
+
+      // Parse the csv file into a json object
+      Papa.parse(csvFile, {
+        delimiter: this.csvDelimiter,
+        header: true,
+        dynamicTyping: true,
+        error: (error) => {
+          this.logger.error(error)
+          reject()
+        },
+        step: (step) => {
+          csvDataParsed.push(step.data)
+        },
+        complete: () => {
+          resolve(csvDataParsed)
+        },
+      })
+    })
+  }
+
+  /**
+   * @param {Object} csvFileInJson - json object of the csv file
+   * @return {void}
+   */
+  convertToHttpBody(csvFileInJson) {
+    const httpBody = []
+    // Reset the buffer for error
+    convertionErrorBuffer.reset()
```

Comment:
I've just noticed that this variable is declared globally but used locally. Please use the same logic than httpBody : return both from the function and declare another local value in the handleFile function like you did for httpBody

Reply:
And this way, you won't need a reset function

Reply:
The only issue is the use of the variable `convertionErrorBuffer` in the function handleFile(), we should change the response in the function `convertToHttpBody`